torch/cuda/__init__.py

```diff
@@ -316,23 +330,24 @@ def stream(stream):
         match the stream.
     """
     if stream is None:
-        yield
+        #yield
         return
-    src_prev_stream = current_stream()
-
-    if src_prev_stream.device != stream.device:
+    #src_prev_stream = current_stream()
+    src_prev_stream = get_current_stream()
```

Comment:
For this to work in JIT, `def stream(...):` needs to return a class with `__enter__` and `__exit__` defined. User code will look like this:
```
with torch.cuda.stream(...):
    ...
```
`with x:` calls `__enter__` and `__exit__` on the object returned by `x`, which in this case is whatever `torch.cuda.stream(...)` returns.